### PR TITLE
First pass at re-enabling tests post-PBE. This will be iterative but

### PR DESCRIPTION
**Pull request by kcibul**
… all moving in the right direction. Test Infrastructure updated to handle the test re-enabled here, including ThreeStep

There are a few intermittent test failures, bugged as #850 #876 #877

### CODE REVIEW

**Comment by kcibul** [comment]
@jmthibault79 @kshakir -- can you review?

**Comment by kcibul** [comment]
please ignore that mention @jmthibault79 -- @mcovarr has it (and I meant @Horneth anyway!)

File: engine/src/test/scala/cromwell/ArrayAndMapIndexingWorkflowSpec.scala [hunk 1]
```diff
@@ -10,10 +10,10 @@ import scala.language.postfixOps
 
 class ArrayAndMapIndexingWorkflowSpec extends CromwellTestkitSpec {
   "A workflow with array/map indexes in expressions" should {
-    "run locally" ignore {
+    "run locally" in {
       runWdlAndAssertOutputs(
         sampleWdl = SampleWdl.ArrayAndMapIndexingWdl,
-        eventFilter = EventFilter.info(pattern = s"starting calls: test.echo_int, test.echo_str", occurrences = 1),
+        eventFilter = EventFilter.info(pattern = "Workflow complete", occurrences = 1),
```
Comment (mcovarr) [inline engine/src/test/scala/cromwell/ArrayAndMapIndexingWorkflowSpec.scala]: I'm not sure this particular filter has much value any more, but currently the API appears to require a filter.

File: engine/src/test/scala/cromwell/CromwellTestkitSpec.scala [hunk 2]
```diff
@@ -299,30 +304,30 @@ abstract class CromwellTestkitSpec extends TestKit(new CromwellTestkitSpec.TestW
...
-      val actualArray = output.asInstanceOf[WdlArray]
-      actualArray.value.size should be(expectedArray.value.size)
-      (actualArray.value zip expectedArray.value) foreach {
-        case (actual, expected) => validateOutput(actual, expected)
-      }
+  // output received from the metadata service is untyped, so WdlValues are converted to strings
+  private def validateOutput(output: String, expectedOutput: WdlValue): Unit = expectedOutput match {
+    case expectedFile: WdlFile =>
+      output.endsWith(expectedFile.valueString) shouldEqual true
+      // TODO: when we re-enable a test in the PBE world that deals with arrays, this will need to be re-instated and refactored
```
Comment (mcovarr) [inline engine/src/test/scala/cromwell/CromwellTestkitSpec.scala]: we've been using `// PBE:` for PBE specific todos

File: engine/src/test/scala/cromwell/CromwellTestkitSpec.scala [hunk 3]
```diff
@@ -156,31 +160,32 @@ object CromwellTestkitSpec {
...
-      Await.result(manager.ask(message).mapTo[WorkflowManagerWorkflowStdoutStderrSuccess], Duration.Inf).logs
+      throw new UnsupportedOperationException()
     }
 
+    @deprecated("should no longer be on WorkflowManagerActor in a PBE world")
     def callStdoutStderr(id: WorkflowId, fqn: FullyQualifiedName): Seq[CallLogs] = {
-      val message = CallStdoutStderr(id, fqn)
-      Await.result(manager.ask(message).mapTo[WorkflowManagerCallStdoutStderrSuccess], Duration.Inf).logs
+      throw new UnsupportedOperationException()
     }
```
Comment (mcovarr) [inline engine/src/test/scala/cromwell/CromwellTestkitSpec.scala]: all these deprecated methods could be one-lined

**Comment by mcovarr** [comment]
:+1: nice bump in the test coverage!  😄

**Comment by kshakir** [comment]
LGTM 👍

**Pull request merged**